toast/tod/sim_detdata.py

```diff
@@ -24,6 +24,82 @@
 from .. import rng as rng
 
 
+def sim_noise_timestream(realization, stream, rate, samples, oversample, freq, psd):
+    
+    fftlen = 2
+    while fftlen <= (oversample * samples):
+        fftlen *= 2
+    half = int(fftlen / 2)
+    norm = rate * float(half)
+    df = rate / fftlen
+
+    interp_freq = np.linspace(df, df*half, num=half, endpoint=True)
+    loginterp_freq = np.log10(interp_freq)
+
+    # ignore zero frequency
+
+    trimzero = False
+    if freq[0] <= 0.0:
+        trimzero = True
+
```

Comment:
What is "freq" includes the negative frequencies? Rather than cropping just the zero frequency, we could discard all non-positive frequencies:
good = freq > 0
rawfreq = freq[good]